scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt

```diff
@@ -159,6 +160,20 @@ abstract class AbstractStorageFunTest : WordSpec() {
                 readResult should beSuccess()
                 (readResult as Success).result should beEmpty()
             }
+
+            "not store a result for the same scanner and provenance twice" {
+                val scanResult = ScanResult(provenanceWithSourceArtifact1, scannerDetails1, scanSummaryWithFiles)
+
+                val addResult1 = storage.add(id1, scanResult)
+                val addResult2 = storage.add(id1, scanResult)
```

Comment:
Would it make sense to use a different summary (non-key attributes) for the second add call, and then assert which of the two scan results is in storage after execution. That'd document the behavior I asked about below.

Reply:
I'll wait for the outcome of the other discussion.

Reply:
I have updated the test.